src/python/pants/backend/docker/target_types.py

```diff
@@ -46,6 +47,24 @@ class DockerBuildArgsField(StringSequenceField):
     )
 
 
+class DockerImageContextRootField(StringField):
+    alias = "context_root"
+    help = (
+        "Specify which directory to use as the Docker build context root, relative to the build "
+        "root by default. With a `./` prefix it is relative to the directory of the BUILD file. "
+        "This affects the file paths to use for the COPY and ADD instructions.\n\nThe "
+        "default build root is specified in the `[docker].default_context_root` configuration "
+        "option."
+    )
+
+    @classmethod
+    def compute_value(cls, raw_value: Optional[str], address: Address) -> Optional[str]:
+        value_or_default = super().compute_value(raw_value, address=address)
+        if isinstance(value_or_default, str):
+            return value_or_default.strip("/")
```

Comment:
I don't understand this?

Reply:
I don't want absolute paths for the context root value, and `/src/...` or `//src/...` etc will be the same as just `src/...` (matching the semantics of target addresses).

So stripping leading `/` wherever they are introduced rather than when referenced. (hence, there's also a `strip("/")` for the `default_context_root` option.)

Reply:
Why support `/` and `//` at all? We don't support that for the `sources` field etc, or CLI specs where you say file paths.

Reply:
True. Could err on abs paths. I had that at first, can't recall now why I changed it..